src/rgw/services/svc_topic_rados.cc

```diff
@@ -0,0 +1,206 @@
+#include "svc_topic_rados.h"
+#include "rgw_notify.h"
+#include "rgw_tools.h"
+#include "rgw_zone.h"
+#include "svc_meta.h"
+#include "svc_meta_be_sobj.h"
+#include "svc_zone.h"
+
+#define dout_subsys ceph_subsys_rgw
+
+static std::string topic_oid_prefix = "topic.";
+static constexpr char topic_tenant_delim[] = ":";
+static std::string bucket_topic_oid_prefix = "buckets.";
+
+std::string get_topic_key(const std::string& topic_name,
+                          const std::string& tenant) {
+  if (tenant.empty()) {
+    return topic_name;
+  }
+  return tenant + topic_tenant_delim + topic_name;
+}
+
+void parse_topic_entry(const std::string& topic_entry,
+                       std::string* tenant_name,
+                       std::string* topic_name) {
+  // expected format: [tenant_name:]topic_name*
+  auto pos = topic_entry.find(topic_tenant_delim);
+  if (pos != std::string::npos) {
+    *tenant_name = topic_entry.substr(0, pos);
+    *topic_name = topic_entry.substr(pos + 1);
+  } else {
+    tenant_name->clear();
+    *topic_name = topic_entry;
+  }
+}
+
+std::string get_bucket_topic_mapping_oid(const rgw_pubsub_topic& topic) {
+  return bucket_topic_oid_prefix + get_topic_key(topic.name, topic.user.tenant);
+}
+
+class RGWSI_Topic_Module : public RGWSI_MBSObj_Handler_Module {
+  RGWSI_Topic_RADOS::Svc& svc;
+  const std::string prefix;
+
+ public:
+  RGWSI_Topic_Module(RGWSI_Topic_RADOS::Svc& _svc)
+      : RGWSI_MBSObj_Handler_Module("topic"),
+        svc(_svc),
+        prefix(topic_oid_prefix) {}
+
+  void get_pool_and_oid(const std::string& key,
+                        rgw_pool* pool,
+                        std::string* oid) override {
+    if (pool) {
+      *pool = svc.zone->get_zone_params().log_pool;
```

Comment:
the topic metadata and topic/bucket mapping objects should go in the same `.rgw.meta` pool as other metadata objects

for example, the `RGWZoneParams::roles_pool` lives in the `roles` namespace of the pool, ie `.rgw.meta:roles`. we probably want a new `topics_pool = ".rgw.meta:topics"`

Reply:
so now the topic and mapping will live in meta pool, but the queue that stores the persistent notification will stay in log pool under `notif` namespace, so we wanna continue that or wanna change the persistent queue location as well ?
originally i kept in log pool coz the v1 objects were stored in log pool

Reply:
we still want the persistent queue in the log pool, but the v2 metadata objects (topics and their bucket listings) should move to the metadata pool to be consistent with other types of metadata

Reply:
update the code to use meta pool for topic n omap